### PR TITLE
[JENKINS-49491] Adapting JTH to a recent version of HtmlUnit

### PR DESCRIPTION
**Pull request by recena**
[JENKINS-49491](https://issues.jenkins-ci.org/browse/JENKINS-49491)

Downstream of https://github.com/jenkinsci/jenkins-test-harness-htmlunit/pull/1

@jenkinsci/code-reviewers

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/HudsonTestCase.java [hunk 1]
```diff
@@ -1397,29 +1397,6 @@ public boolean processSynchron(HtmlPage page, WebRequest settings, boolean async
                 }
             });
 
-            setCssErrorHandler(new ErrorHandler() {
```
Comment (recena) [inline src/main/java/org/jvnet/hudson/test/HudsonTestCase.java]: I need to replace this one.

**Comment by recena** [comment]
Why do not we remove `HudsonTestCase.java`?

**Comment by MarkEWaite** [comment]
@recena I thought that removing the `HudsonTestCase` class would break those plugins which have not yet migrated from HudsonTestCase to JenkinsRule.  The git plugin only made that transition in 2016, so I assume there are other plugins which still depend on HudsonTestCase.

**Comment by recena** [comment]
@MarkEWaite For those cases, the maintainer has to make a decision. If he wants to update `JTH` then needs to review his tests. Or do you mean about OSS ATH?

**Comment by jglick** [comment]
Yes there are a _lot_ of tests out there still extending `HudsonTestCase`. While we would certainly like to continue to update them, it is labor-intensive, so anything which really breaks it had better have a very solid justification. Merely updating HtmlUnit is not enough reason.

**Comment by recena** [comment]
The motivation is not relared with upgrading HtmlUnit but with keeping the things clearer.

**Comment by rtyler** [comment]
FWIW, without updating HTMLUnit, it's not possible (as far as I can tell) to use ECMAScript 6, or anything moderately considered "contemporary" JavaScript in Jenkins Core.

IMHO that's a strong motivation.

**Comment by recena** [comment]
@rtyler I could continue working here whether it is still interesting for the community.

**Comment by uhafner** [comment]
Fixed with https://github.com/jenkinsci/jenkins-test-harness/pull/103

**Pull request closed without merging**